core/src/main/scala/org/apache/spark/deploy/history/ApplicationHistoryProvider.scala

```diff
@@ -76,6 +76,14 @@ private[history] case class LoadedAppUI(
 private[history] abstract class ApplicationHistoryProvider {
 
   /**
+   * The number of applications available for listing. Separate method in case it's cheaper
+   * to get a count than to calculate the whole listing.
```

Comment:
This is an interface, so this was added to allow implementations to override this method if that makes sense.

It just looks like I lost the override in one of my rebases, so let me add that back.